## This patch adds soname version numbers
## MRIBiasFieldCorrection is using code from review, cant compile it

Index: slicer-3.6.3~svn16592/Applications/CLI/MIDASApplications/DownloadSlicerTutorials/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Applications/CLI/MIDASApplications/DownloadSlicerTutorials/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Applications/CLI/MIDASApplications/DownloadSlicerTutorials/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -18,8 +18,8 @@
 
 #
 # Find SLICERLIBCURL (slicercurl)
-find_package(SLICERLIBCURL REQUIRED)
-include(${SLICERLIBCURL_USE_FILE})
+find_package(CURL REQUIRED)
+include_directories(${CURL_INCLUDE_DIRS})
 
 #
 # Find GenerateCLP
@@ -42,7 +42,7 @@
   COMPILE_FLAGS "-Dmain=ModuleEntryPoint")
 slicer3_set_plugins_output_path(DownloadSlicerTutorialsModule)
 target_link_libraries(DownloadSlicerTutorialsModule 
-  ${SLICERLIBCURL_LIBRARIES} ITKIO MRMLIDIO itkzlib)
+  ${CURL_LIBRARIES} ITKIO MRMLIDIO)
 
 #
 # Generate executable
Index: slicer-3.6.3~svn16592/Applications/CLI/MIDASApplications/DownloadSlicerTutorials/DownloadSlicerTutorials.cxx
===================================================================
--- slicer-3.6.3~svn16592.orig/Applications/CLI/MIDASApplications/DownloadSlicerTutorials/DownloadSlicerTutorials.cxx	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Applications/CLI/MIDASApplications/DownloadSlicerTutorials/DownloadSlicerTutorials.cxx	2011-05-09 23:31:20.000000000 -0400
@@ -3,7 +3,7 @@
 #endif
 
 // Includes
-#include <slicerlibcurl/slicerlibcurl.h>
+#include <curl/curl.h>
 #include <itksys/SystemTools.hxx>
 #include <itksys/Process.h>
 #include <math.h>
Index: slicer-3.6.3~svn16592/Applications/GUI/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Applications/GUI/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Applications/GUI/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -45,7 +45,7 @@
     set(Slicer3_FORWARD_PATH_BUILD 
       "\"${Slicer3_FORWARD_DIR_BUILD}\",\"${VTK_DIR}/bin\",\"${KWWidgets_RUNTIME_DIRS}\",\"${ITK_DIR}/bin\",\"${Teem_DIR}/bin\",\"${TCL_DIR}/bin\",\"${INCR_TCL_LIBRARY_PATH}\",\"${INCR_TK_LIBRARY_PATH}\",\"${OPENGL_gl_LIBRARY_PATH}\"")
     set(Slicer3_FORWARD_PATH_INSTALL
-      "\"../${Slicer3_INSTALL_LIB_DIR}\",\"../lib/vtk-${VTK_MAJOR_VERSION}.${VTK_MINOR_VERSION}\",\"../lib/KWWidgets\",\"../lib/InsightToolkit\",\"../lib/Teem-${Teem_VERSION_MAJOR}.${Teem_VERSION_MINOR}\",\"../lib/TclTk/lib\",\"../lib/TclTk/lib/itcl${INCR_TCL_VERSION_DOT}\",\"../lib/TclTk/lib/itk${INCR_TCL_VERSION_DOT}\",\"../lib/FreeSurfer\",\"../lib/IGT\",\"../lib/ITKCommandIO\",\"../lib/LoadableModule\",\"../lib/MGHImageIO\",\"../lib/MRML\",\"../lib/ModuleDescriptionParser\",\"../lib/MRMLIDImageIO\",\"../lib/Qdec\",\"../lib/RemoteIO\",\"../lib/SlicerIO\",\"../lib/tclap\",\"../lib/vtkITK\",\"../lib/vtkTeem\"")
+      "\"../${Slicer3_INSTALL_LIB_DIR}\",\"../lib/vtk-${VTK_MAJOR_VERSION}.${VTK_MINOR_VERSION}\"")
     if(Slicer3_BUILD_MODULES)
       set(Slicer3_FORWARD_PATH_BUILD 
         "${Slicer3_FORWARD_PATH_BUILD},\"${Slicer3_HOME}/${Slicer3_INSTALL_MODULES_BIN_DIR}\"")
@@ -95,8 +95,10 @@
 # --------------------------------------------------------------------------
 # Find SLICERLIBCURL (slicercurl)
 #
-find_package(SLICERLIBCURL REQUIRED)
-include(${SLICERLIBCURL_USE_FILE})
+#find_package(SLICERLIBCURL REQUIRED)
+#include(${SLICERLIBCURL_USE_FILE})
+find_package(CURL REQUIRED)
+include_directories(${CURL_INCLUDE_DIRS})
 
 # --------------------------------------------------------------------------
 # Include dirs
Index: slicer-3.6.3~svn16592/Applications/GUI/Slicer3.cxx
===================================================================
--- slicer-3.6.3~svn16592.orig/Applications/GUI/Slicer3.cxx	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Applications/GUI/Slicer3.cxx	2011-05-09 23:31:20.000000000 -0400
@@ -449,7 +449,7 @@
     }
 
   std::string tclEnv = "TCL_LIBRARY=";
-  tclEnv += slicerHome + "/lib/TclTk/lib/tcl" + Slicer3_TCL_TK_MAJOR_VERSION + "." + Slicer3_TCL_TK_MINOR_VERSION;
+  tclEnv += slicerHome + "/lib/tcl" + Slicer3_TCL_TK_MAJOR_VERSION + "." + Slicer3_TCL_TK_MINOR_VERSION;
   vtkKWApplication::PutEnv(const_cast <char *> (tclEnv.c_str()));
 
   // ITK factories dir
Index: slicer-3.6.3~svn16592/Libs/FreeSurfer/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/FreeSurfer/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/FreeSurfer/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -95,7 +95,7 @@
 
 install(TARGETS ${lib_name}
   RUNTIME DESTINATION bin COMPONENT RuntimeLibraries 
-  LIBRARY DESTINATION lib/${PROJECT_NAME} COMPONENT RuntimeLibraries
+  LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
   ARCHIVE DESTINATION lib/${PROJECT_NAME} COMPONENT Development
   )
 
@@ -125,9 +125,22 @@
     SET_TARGET_PROPERTIES(${lib_name}Python PROPERTIES SUFFIX ".pyd")
   ENDIF(WIN32 AND NOT CYGWIN)
 
+
+  # Apply user-defined properties to the library target.
+  IF(Slicer3_LIBRARY_PROPERTIES)
+    SET_TARGET_PROPERTIES(${lib_name}Python PROPERTIES
+       ${Slicer3_LIBRARY_PROPERTIES}
+       LINK_INTERFACE_LIBRARIES ""
+    )
+    SET_TARGET_PROPERTIES(${lib_name}PythonD PROPERTIES
+       ${Slicer3_LIBRARY_PROPERTIES}
+       LINK_INTERFACE_LIBRARIES ""
+    )
+  ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
   INSTALL(TARGETS ${lib_name}PythonD ${lib_name}Python
     RUNTIME DESTINATION bin COMPONENT RuntimeLibraries 
-    LIBRARY DESTINATION lib/${PROJECT_NAME} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION lib/${PROJECT_NAME} COMPONENT Development)
 
 ENDIF(VTK_WRAP_PYTHON AND BUILD_SHARED_LIBS)
Index: slicer-3.6.3~svn16592/Libs/IGT/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/IGT/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/IGT/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -157,7 +157,7 @@
 
 install(TARGETS ${lib_name}
   RUNTIME DESTINATION bin COMPONENT RuntimeLibraries 
-  LIBRARY DESTINATION lib/${PROJECT_NAME} COMPONENT RuntimeLibraries
+  LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
   ARCHIVE DESTINATION lib/${PROJECT_NAME} COMPONENT Development
   )
 
Index: slicer-3.6.3~svn16592/Libs/ITKCommandIO/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/ITKCommandIO/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/ITKCommandIO/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -75,7 +75,7 @@
 
 install(TARGETS ${lib_name}  
   RUNTIME DESTINATION bin COMPONENT RuntimeLibraries
-  LIBRARY DESTINATION lib/${PROJECT_NAME} COMPONENT RuntimeLibraries
+  LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
   ARCHIVE DESTINATION lib/${PROJECT_NAME} COMPONENT Development
   )
 
Index: slicer-3.6.3~svn16592/Libs/LoadableModule/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/LoadableModule/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/LoadableModule/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -127,7 +127,7 @@
 
 install(TARGETS ${lib_name} 
   RUNTIME DESTINATION bin COMPONENT RuntimeLibraries
-  LIBRARY DESTINATION lib/${PROJECT_NAME} COMPONENT RuntimeLibraries
+  LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
   ARCHIVE DESTINATION lib/${PROJECT_NAME} COMPONENT Development
   )
 
Index: slicer-3.6.3~svn16592/Libs/MGHImageIO/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/MGHImageIO/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/MGHImageIO/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -63,7 +63,7 @@
 
 install(TARGETS ${lib_name} 
   RUNTIME DESTINATION bin COMPONENT RuntimeLibraries
-  LIBRARY DESTINATION lib/${PROJECT_NAME} COMPONENT RuntimeLibraries
+  LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
   ARCHIVE DESTINATION lib/${PROJECT_NAME} COMPONENT Development
   )
 
Index: slicer-3.6.3~svn16592/Libs/MRML/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/MRML/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/MRML/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -268,7 +268,7 @@
 
 install(TARGETS ${lib_name}
   RUNTIME DESTINATION bin COMPONENT RuntimeLibraries 
-  LIBRARY DESTINATION lib/${PROJECT_NAME} COMPONENT RuntimeLibraries
+  LIBRARY DESTINATION lib/Slicer3 COMPONENT RuntimeLibraries
   ARCHIVE DESTINATION lib/${PROJECT_NAME} COMPONENT Development
   )
 
@@ -298,9 +298,21 @@
     SET_TARGET_PROPERTIES(${lib_name}Python PROPERTIES SUFFIX ".pyd")
   ENDIF(WIN32 AND NOT CYGWIN)
 
+  # Apply user-defined properties to the library target.
+  IF(Slicer3_LIBRARY_PROPERTIES)
+    SET_TARGET_PROPERTIES(${lib_name}Python PROPERTIES
+       ${Slicer3_LIBRARY_PROPERTIES}
+       LINK_INTERFACE_LIBRARIES ""
+    )
+    SET_TARGET_PROPERTIES(${lib_name}PythonD PROPERTIES
+       ${Slicer3_LIBRARY_PROPERTIES}
+       LINK_INTERFACE_LIBRARIES ""
+    )
+  ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
   INSTALL(TARGETS ${lib_name}PythonD ${lib_name}Python
     RUNTIME DESTINATION bin COMPONENT RuntimeLibraries 
-    LIBRARY DESTINATION lib/${PROJECT_NAME} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION lib/${PROJECT_NAME} COMPONENT Development)
 
 ENDIF(VTK_WRAP_PYTHON AND BUILD_SHARED_LIBS)
Index: slicer-3.6.3~svn16592/Libs/MRMLIDImageIO/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/MRMLIDImageIO/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/MRMLIDImageIO/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -85,7 +85,7 @@
 
 install(TARGETS ${lib_name} 
   RUNTIME DESTINATION bin COMPONENT RuntimeLibraries
-  LIBRARY DESTINATION lib/${PROJECT_NAME} COMPONENT RuntimeLibraries
+  LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
   ARCHIVE DESTINATION lib/${PROJECT_NAME} COMPONENT Development
   )
 
Index: slicer-3.6.3~svn16592/Libs/RemoteIO/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/RemoteIO/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/RemoteIO/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -13,8 +13,11 @@
 find_package(ITK REQUIRED)
 include(${ITK_USE_FILE})
 
-find_package(SLICERLIBCURL REQUIRED)
-include(${SLICERLIBCURL_USE_FILE})
+find_package(CURL REQUIRED)
+if(CURL_FOUND)
+  # Add include directories needed to use CURL.
+  include_directories(${CURL_INCLUDE_DIRS})
+endif(CURL_FOUND)
 
 # Disable MSVC 8 warnings
 if(WIN32)
@@ -108,7 +111,7 @@
   )
   
 SET(libs
-  ${SLICERLIBCURL_LIBRARIES}
+  ${CURL_LIBRARIES}
   ${VTK_LIBRARIES}
   )
 
@@ -123,7 +126,7 @@
 
 install(TARGETS ${lib_name}
   RUNTIME DESTINATION bin COMPONENT RuntimeLibraries 
-  LIBRARY DESTINATION lib/${PROJECT_NAME} COMPONENT RuntimeLibraries
+  LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
   ARCHIVE DESTINATION lib/${PROJECT_NAME} COMPONENT Development
   )
 
@@ -153,9 +156,22 @@
     SET_TARGET_PROPERTIES(${lib_name}Python PROPERTIES SUFFIX ".pyd")
   ENDIF(WIN32 AND NOT CYGWIN)
 
+  # Apply user-defined properties to the library target.
+  IF(Slicer3_LIBRARY_PROPERTIES)
+    SET_TARGET_PROPERTIES(${lib_name}Python PROPERTIES
+       ${Slicer3_LIBRARY_PROPERTIES}
+       LINK_INTERFACE_LIBRARIES ""
+    )
+    SET_TARGET_PROPERTIES(${lib_name}PythonD PROPERTIES
+       ${Slicer3_LIBRARY_PROPERTIES}
+       LINK_INTERFACE_LIBRARIES ""
+   )
+  ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
+
   INSTALL(TARGETS ${lib_name}PythonD ${lib_name}Python
     RUNTIME DESTINATION bin COMPONENT RuntimeLibraries 
-    LIBRARY DESTINATION lib/${PROJECT_NAME} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION lib/${PROJECT_NAME} COMPONENT Development)
 
 ENDIF(VTK_WRAP_PYTHON AND BUILD_SHARED_LIBS)
Index: slicer-3.6.3~svn16592/Libs/RemoteIO/vtkHTTPHandler.h
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/RemoteIO/vtkHTTPHandler.h	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/RemoteIO/vtkHTTPHandler.h	2011-05-09 23:31:20.000000000 -0400
@@ -5,7 +5,7 @@
 #pragma warning ( disable : 4786 )
 #endif
 
-#include <slicerlibcurl/slicerlibcurl.h>
+#include <curl/curl.h>
 
 #include <vtkRemoteIOConfigure.h>
 #include "vtkRemoteIO.h"
Index: slicer-3.6.3~svn16592/Libs/SlicerIO/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/SlicerIO/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/SlicerIO/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -50,7 +50,7 @@
 
 install(TARGETS ${lib_name}
   RUNTIME DESTINATION bin COMPONENT RuntimeLibraries 
-  LIBRARY DESTINATION lib/${PROJECT_NAME} COMPONENT RuntimeLibraries
+  LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
   ARCHIVE DESTINATION lib/${PROJECT_NAME} COMPONENT Development
   )
 
Index: slicer-3.6.3~svn16592/Libs/vtkITK/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/vtkITK/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/vtkITK/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -75,7 +75,6 @@
   vtkITKBSplineTransform.cxx
   vtkITKTimeSeriesDatabase.cxx
   vtkITKIslandMath.cxx
-  vtkITKGrowCutSegmentationImageFilter.cxx
   )
 
 # Abstract/pure virtual classes
@@ -150,9 +149,14 @@
   )
 ENDIF(Slicer3_LIBRARY_PROPERTIES)
 
+#set_target_properties(${lib_name}
+#  PROPERTIES
+#  LINK_INTERFACE_LIBRARIES ""
+#)
+
 install(TARGETS ${lib_name}
   RUNTIME DESTINATION bin COMPONENT RuntimeLibraries 
-  LIBRARY DESTINATION lib/${PROJECT_NAME} COMPONENT RuntimeLibraries
+  LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
   ARCHIVE DESTINATION lib/${PROJECT_NAME} COMPONENT Development
   )
 
Index: slicer-3.6.3~svn16592/Libs/vtkTeem/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/vtkTeem/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/vtkTeem/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -136,7 +136,7 @@
 
 install(TARGETS ${lib_name}
   RUNTIME DESTINATION bin COMPONENT RuntimeLibraries 
-  LIBRARY DESTINATION lib/${PROJECT_NAME} COMPONENT RuntimeLibraries
+  LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
   ARCHIVE DESTINATION lib/${PROJECT_NAME} COMPONENT Development
   )
 
@@ -166,9 +166,21 @@
     SET_TARGET_PROPERTIES(${lib_name}Python PROPERTIES SUFFIX ".pyd")
   ENDIF(WIN32 AND NOT CYGWIN)
 
+  # Apply user-defined properties to the library target.
+  IF(Slicer3_LIBRARY_PROPERTIES)
+    SET_TARGET_PROPERTIES(${lib_name}Python PROPERTIES
+       ${Slicer3_LIBRARY_PROPERTIES}
+       LINK_INTERFACE_LIBRARIES ""
+    )
+    SET_TARGET_PROPERTIES(${lib_name}PythonD PROPERTIES
+       ${Slicer3_LIBRARY_PROPERTIES}
+       LINK_INTERFACE_LIBRARIES ""
+    )
+  ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
   INSTALL(TARGETS ${lib_name}PythonD ${lib_name}Python
     RUNTIME DESTINATION bin COMPONENT RuntimeLibraries 
-    LIBRARY DESTINATION lib/${PROJECT_NAME} COMPONENT RuntimeLibraries
+    LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
     ARCHIVE DESTINATION lib/${PROJECT_NAME} COMPONENT Development)
 
 ENDIF(VTK_WRAP_PYTHON AND BUILD_SHARED_LIBS)
Index: slicer-3.6.3~svn16592/Modules/ChangeTracker/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/ChangeTracker/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/ChangeTracker/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -44,8 +44,6 @@
 
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 install(FILES 
-  ${headers} 
-  "${CMAKE_CURRENT_BINARY_DIR}/vtkChangeTrackerConfigure.h"
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
   )
 
Index: slicer-3.6.3~svn16592/Modules/CommandLineModule/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/CommandLineModule/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/CommandLineModule/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -33,8 +33,6 @@
 
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 install(FILES 
-  ${headers} 
-  "${CMAKE_CURRENT_BINARY_DIR}/vtkCommandLineModuleConfigure.h"
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
   )
 
Index: slicer-3.6.3~svn16592/Modules/FetchMI/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/FetchMI/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/FetchMI/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -44,8 +44,6 @@
 
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 install(FILES 
-  ${headers} 
-  "${CMAKE_CURRENT_BINARY_DIR}/vtkFetchMIConfigure.h"
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
   )
 
@@ -93,6 +91,13 @@
   Volumes
   )
 
+# Apply user-defined properties to the library target.
+IF(Slicer3_LIBRARY_PROPERTIES)
+  SET_TARGET_PROPERTIES(${lib_name} PROPERTIES
+    ${Slicer3_LIBRARY_PROPERTIES}
+  )
+ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
 slicer3_install_modules(${lib_name})
 
 # --------------------------------------------------------------------------
Index: slicer-3.6.3~svn16592/Modules/GradientAnisotropicDiffusionFilter/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/GradientAnisotropicDiffusionFilter/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/GradientAnisotropicDiffusionFilter/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -41,8 +41,6 @@
 
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 install(FILES 
-  ${headers} 
-  "${CMAKE_CURRENT_BINARY_DIR}/vtkGradientAnisotropicDiffusionFilterConfigure.h"
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
   )
 
@@ -106,6 +104,13 @@
   ${ITK_LIBRARIES}
   )
 
+# Apply user-defined properties to the library target.
+IF(Slicer3_LIBRARY_PROPERTIES)
+  SET_TARGET_PROPERTIES(${lib_name} PROPERTIES
+    ${Slicer3_LIBRARY_PROPERTIES}
+  )
+ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
 slicer3_install_modules(${lib_name})
 
 # --------------------------------------------------------------------------
Index: slicer-3.6.3~svn16592/Modules/Meshing/IA_FEMesh/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/Meshing/IA_FEMesh/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/Meshing/IA_FEMesh/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -46,8 +46,6 @@
 
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 install(FILES 
-  ${headers} 
-  "${CMAKE_CURRENT_BINARY_DIR}/vtkIA_FEMeshConfigure.h"
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
   )
   
@@ -104,4 +102,11 @@
   BuildingBlock
   )
 
+# Apply user-defined properties to the library target.
+IF(Slicer3_LIBRARY_PROPERTIES)
+  SET_TARGET_PROPERTIES(${LIB_NAME} PROPERTIES
+    ${Slicer3_LIBRARY_PROPERTIES}
+  )
+ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
 slicer3_install_modules(${LIB_NAME})
Index: slicer-3.6.3~svn16592/Modules/NeuroNav/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/NeuroNav/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/NeuroNav/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -41,8 +41,6 @@
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 
 install(FILES 
-  ${headers} 
-  "${CMAKE_CURRENT_BINARY_DIR}/vtkNeuroNavConfigure.h"
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
   )
 
@@ -89,6 +87,13 @@
   ${ITK_LIBRARIES}
   )
 
+# Apply user-defined properties to the library target.
+IF(Slicer3_LIBRARY_PROPERTIES)
+  SET_TARGET_PROPERTIES(${lib_name} PROPERTIES
+    ${Slicer3_LIBRARY_PROPERTIES}
+  )
+ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
 slicer3_install_modules(${lib_name})
 
 
Index: slicer-3.6.3~svn16592/Modules/OpenIGTLinkIF/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/OpenIGTLinkIF/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/OpenIGTLinkIF/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -42,8 +42,6 @@
   )
 
 install(FILES 
-  ${headers} 
-  "${CMAKE_CURRENT_BINARY_DIR}/vtkOpenIGTLinkIFConfigure.h"
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
   )
 
@@ -95,6 +93,14 @@
   ${ITK_LIBRARIES}
   ${OpenIGTLink_LIBRARIES}
   )
+
+# Apply user-defined properties to the library target.
+IF(Slicer3_LIBRARY_PROPERTIES)
+  SET_TARGET_PROPERTIES(${lib_name} PROPERTIES
+    ${Slicer3_LIBRARY_PROPERTIES}
+  )
+ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
 slicer3_install_modules(${lib_name})
 
 # --------------------------------------------------------------------------
Index: slicer-3.6.3~svn16592/Modules/ProstateNav/Wizard/vtkProstateNavCalibrationStep.h
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/ProstateNav/Wizard/vtkProstateNavCalibrationStep.h	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/ProstateNav/Wizard/vtkProstateNavCalibrationStep.h	2011-05-09 23:31:20.000000000 -0400
@@ -31,6 +31,13 @@
 class VTK_PROSTATENAV_EXPORT vtkProstateNavCalibrationStep : public vtkProstateNavStep
 {
 public:
+  //BTX
+  typedef NEWMAT::Matrix Matrix;
+  typedef NEWMAT::Real Real;
+  typedef NEWMAT::FFT2 FFT2;
+  typedef NEWMAT::FFT2I FFT2I;
+  //ETX
+
   static vtkProstateNavCalibrationStep *New();
   vtkTypeRevisionMacro(vtkProstateNavCalibrationStep,vtkProstateNavStep);
   void PrintSelf(ostream& os, vtkIndent indent);
Index: slicer-3.6.3~svn16592/Modules/QueryAtlas/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/QueryAtlas/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/QueryAtlas/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -42,14 +42,11 @@
 
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 install(FILES 
-  ${headers} 
-  "${CMAKE_CURRENT_BINARY_DIR}/vtkQueryAtlasConfigure.h"
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
   )
 
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/Resources/*.h")
 install(FILES 
-  ${headers} 
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME}/Resources COMPONENT Development
   ) 
 
@@ -110,6 +107,13 @@
   ${ITK_LIBRARIES}
   )
 
+# Apply user-defined properties to the library target.
+IF(Slicer3_LIBRARY_PROPERTIES)
+  SET_TARGET_PROPERTIES(${lib_name} PROPERTIES
+    ${Slicer3_LIBRARY_PROPERTIES}
+  )
+ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
 slicer3_install_modules(${lib_name})
 
 # --------------------------------------------------------------------------
@@ -203,14 +207,14 @@
   OntologyViz/extjars/prefuse.jar    
   )
 
-foreach(file ${ONTOVIZFILES})
-  configure_file(
-    ${CMAKE_CURRENT_SOURCE_DIR}/${file}
-    ${CMAKE_BINARY_DIR}/${Slicer3_INSTALL_MODULES_LIB_DIR}/${PROJECT_NAME}/${file}
-    COPYONLY)
-  get_filename_component(path "${file}" PATH)
-  install(FILES 
-    ${file} 
-    DESTINATION ${Slicer3_INSTALL_MODULES_LIB_DIR}/${PROJECT_NAME}/${path} COMPONENT Development
-    ) 
-endforeach(file)
+#foreach(file ${ONTOVIZFILES})
+#  configure_file(
+#    ${CMAKE_CURRENT_SOURCE_DIR}/${file}
+#    ${CMAKE_BINARY_DIR}/${Slicer3_INSTALL_MODULES_LIB_DIR}/${PROJECT_NAME}/${file}
+#    COPYONLY)
+#  get_filename_component(path "${file}" PATH)
+#  install(FILES
+#    ${file}
+#    DESTINATION ${Slicer3_INSTALL_MODULES_LIB_DIR}/${PROJECT_NAME}/${path} COMPONENT Development
+#    )
+#endforeach(file)
Index: slicer-3.6.3~svn16592/Modules/RealTimeImaging/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/RealTimeImaging/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/RealTimeImaging/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -34,7 +34,6 @@
 
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 install(FILES 
-  ${headers} 
   "${CMAKE_CURRENT_BINARY_DIR}/vtkRealTimeImagingConfigure.h"
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
   )
Index: slicer-3.6.3~svn16592/Modules/ScriptedModule/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/ScriptedModule/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/ScriptedModule/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -33,8 +33,6 @@
 
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 install(FILES 
-  ${headers} 
-  "${CMAKE_CURRENT_BINARY_DIR}/vtkScriptedModuleConfigure.h"
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
   )
 
@@ -71,6 +69,7 @@
   ${ITK_LIBRARIES}
   )
 
+
 # Python settings
 
 if(Slicer3_USE_PYTHON)
Index: slicer-3.6.3~svn16592/Modules/SlicerDaemon/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/SlicerDaemon/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/SlicerDaemon/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -33,8 +33,6 @@
 
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 install(FILES 
-  ${headers} 
-  "${CMAKE_CURRENT_BINARY_DIR}/vtkSlicerDaemonConfigure.h"
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
   )
 
@@ -78,6 +76,7 @@
   )
 ENDIF(Slicer3_LIBRARY_PROPERTIES)
 
+
 slicer3_install_modules(${lib_name})
 
 # --------------------------------------------------------------------------
Index: slicer-3.6.3~svn16592/Modules/SlicerWelcome/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/SlicerWelcome/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/SlicerWelcome/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -39,8 +39,6 @@
 
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 install(FILES 
-  ${headers} 
-  "${CMAKE_CURRENT_BINARY_DIR}/vtkSlicerWelcomeConfigure.h"
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
   )
 
@@ -82,5 +80,12 @@
   ${KWWidgets_LIBRARIES}
   )
 
+# Apply user-defined properties to the library target.
+IF(Slicer3_LIBRARY_PROPERTIES)
+  SET_TARGET_PROPERTIES(${lib_name} PROPERTIES
+    ${Slicer3_LIBRARY_PROPERTIES}
+  )
+ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
 slicer3_install_modules(${lib_name})
 
Index: slicer-3.6.3~svn16592/Modules/Statistics/LabelStatistics/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/Statistics/LabelStatistics/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/Statistics/LabelStatistics/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -33,8 +33,6 @@
 
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 install(FILES 
-  ${headers} 
-  "${CMAKE_CURRENT_BINARY_DIR}/vtkLabelStatisticsConfigure.h"
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
   )
 
@@ -75,6 +73,13 @@
   ${ITK_LIBRARIES}
   )
 
+# Apply user-defined properties to the library target.
+IF(Slicer3_LIBRARY_PROPERTIES)
+  SET_TARGET_PROPERTIES(${lib_name} PROPERTIES
+    ${Slicer3_LIBRARY_PROPERTIES}
+  )
+ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
 slicer3_install_modules(${lib_name})
 
 # --------------------------------------------------------------------------
Index: slicer-3.6.3~svn16592/Modules/Tractography/Display/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/Tractography/Display/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/Tractography/Display/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -33,14 +33,11 @@
 
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 install(FILES 
-  ${headers} 
-  "${CMAKE_CURRENT_BINARY_DIR}/vtkSlicerTractographyDisplayConfigure.h"
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
   )
 
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/Resources/*.h")
 install(FILES 
-  ${headers} 
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME}/Resources COMPONENT Development
   )
 
@@ -87,4 +84,11 @@
   ${ITK_LIBRARIES}
   )
 
+# Apply user-defined properties to the library target.
+IF(Slicer3_LIBRARY_PROPERTIES)
+  SET_TARGET_PROPERTIES(${lib_name} PROPERTIES
+    ${Slicer3_LIBRARY_PROPERTIES}
+  )
+ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
 slicer3_install_modules(${lib_name})
Index: slicer-3.6.3~svn16592/Modules/Tractography/FiducialSeeding/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/Tractography/FiducialSeeding/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/Tractography/FiducialSeeding/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -35,8 +35,6 @@
 
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 install(FILES 
-  ${headers} 
-  "${CMAKE_CURRENT_BINARY_DIR}/vtkSlicerTractographyFiducialSeedingConfigure.h"
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
   )
 
@@ -85,6 +83,7 @@
   )
 ENDIF(Slicer3_LIBRARY_PROPERTIES)
 
+
 slicer3_install_modules(${lib_name})
 
 # --------------------------------------------------------------------------
Index: slicer-3.6.3~svn16592/Modules/VolumeRendering/VolumeRenderingReplacements/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/VolumeRendering/VolumeRenderingReplacements/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/VolumeRendering/VolumeRenderingReplacements/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -33,8 +33,6 @@
 
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 install(FILES 
-  ${headers} 
-  "${CMAKE_CURRENT_BINARY_DIR}/vtkVolumeRenderingReplacementsConfigure.h"
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
   )
 
Index: slicer-3.6.3~svn16592/Modules/Volumes/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/Volumes/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/Volumes/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -48,7 +48,6 @@
 
 file(GLOB headers "${CMAKE_CURRENT_SOURCE_DIR}/*.h")
 install(FILES 
-  ${headers} 
   "${CMAKE_CURRENT_BINARY_DIR}/vtkVolumesConfigure.h"
   DESTINATION ${Slicer3_INSTALL_MODULES_INCLUDE_DIR}/${PROJECT_NAME} COMPONENT Development
   )
Index: slicer-3.6.3~svn16592/Scripts/versioner.tcl
===================================================================
--- slicer-3.6.3~svn16592.orig/Scripts/versioner.tcl	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Scripts/versioner.tcl	2011-05-09 23:31:20.000000000 -0400
@@ -14,15 +14,6 @@
 # Initiated - sp - 2009-03-24
 #
 
-################################################################################
-
-# for subversion repositories (Sandbox)
-if {[info exists ::env(SVN)]} {
-    set ::SVN $::env(SVN)
-} else {
-    set ::SVN svn
-}
-
 
 ################################################################################
 # build the lib/Slicer3/Slicer3version.txt file
@@ -36,31 +27,18 @@
 cd [file dirname [info script]]
 cd ..
 set ::Slicer3_HOME [pwd]
-
-source $::Slicer3_HOME/slicer_variables.tcl
-
+set ::Slicer3_BUILD $::Slicer3_HOME/DEB_build_shared
 set ::Slicer3_BUILDDATE [clock format [clock seconds] -format %Y-%m-%d]
 
 set ::ENV(LANG) "C"
-set svninfo [split [exec svn info] "\n"]
-array set svn ""
-foreach line $svninfo {
-  foreach {tag value} $line {
-    if { $tag == "URL:" } {
-      set svn(URL) $value
-    }
-    if { $tag == "Revision:" } {
-      set svn(revision) $value
-    }
-  }
-}
 cd $cwd
 
 set versionText ""
-set versionText "${versionText}build $::env(BUILD)\n"
+set versionText "${versionText}os $::tcl_platform(os)"
+set versionText "${versionText}machine $::tcl_platform(machine)"
 set versionText "${versionText}buildDate $::Slicer3_BUILDDATE\n"
-set versionText "${versionText}svnurl $svn(URL)\n"
-set versionText "${versionText}svnrevision $svn(revision)\n"
+set versionText "${versionText}svnurl http://svn.slicer.org/Slicer3/trunk\n"
+set versionText "${versionText}svnrevision 16075\n"
 
 set versionFile $::Slicer3_BUILD/lib/Slicer3/Slicer3Version.txt
 puts "Writing version information to $versionFile"
Index: slicer-3.6.3~svn16592/Applications/CLI/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Applications/CLI/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Applications/CLI/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -329,7 +329,7 @@
 slicer3_set_plugins_output_path(${CLP})
 target_link_libraries (${CLP} ${CLP}Lib)
 #####################
-add_subdirectory(DicomToNrrdConverter)
+###add_subdirectory(DicomToNrrdConverter)
 #####################
 # set (CLP DWIDicomLoad)
 
@@ -736,14 +736,14 @@
 ###################
 
 SET(cli_modules
-  MRIBiasFieldCorrection
+  #MRIBiasFieldCorrection
   SparseFieldLevelSetContour
   Realign
   DiffusionApplications
   #EndoNavTracking
   ExtractSkeleton
   #Java
-  N4ITKBiasFieldCorrection
+  #N4ITKBiasFieldCorrection
   RobustStatisticsSegmenter
   )
 
@@ -753,11 +753,11 @@
 
 else ( ${ITK_VERSION_MAJOR} GREATER 3 )
 
-  if ( ${ITK_VERSION_MAJOR} MATCHES 3 AND ${ITK_VERSION_MINOR} GREATER 12 )
+  if ( ${ITK_VERSION_MAJOR} MATCHES 3 AND ${ITK_VERSION_MINOR} GREATER 20 )
 
     SET(cli_modules ${cli_modules} RegisterImagesModule)
 
-  else ( ${ITK_VERSION_MAJOR} MATCHES 3 AND ${ITK_VERSION_MINOR} GREATER 12 )
+  else ( ${ITK_VERSION_MAJOR} MATCHES 3 AND ${ITK_VERSION_MINOR} GREATER 20 )
 
     if ( ${ITK_VERSION_MAJOR} MATCHES 3 AND ${ITK_VERSION_MINOR} GREATER 6 )
       if ( ITK_USE_OPTIMIZED_REGISTRATION_METHODS )
@@ -769,7 +769,7 @@
       message ("RegisterImagesModule requires ITK version 3.8 or higher")
     endif ( ${ITK_VERSION_MAJOR} MATCHES 3 AND ${ITK_VERSION_MINOR} GREATER 6 )
 
-  endif ( ${ITK_VERSION_MAJOR} MATCHES 3 AND ${ITK_VERSION_MINOR} GREATER 12 )
+  endif ( ${ITK_VERSION_MAJOR} MATCHES 3 AND ${ITK_VERSION_MINOR} GREATER 20 )
 
 endif ( ${ITK_VERSION_MAJOR} GREATER 3 )
 
Index: slicer-3.6.3~svn16592/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -29,6 +29,11 @@
 
 set(Slicer3_HOME "${Slicer3_BINARY_DIR}")
 
+
+# TCL
+find_package(TCL REQUIRED)
+
+
 #-----------------------------------------------------------------------------
 # Disable the warnings that DevStudio 2005 emits wrt to sprintf, strcpu, etc.
 #
@@ -295,10 +300,11 @@
 
 #-----------------------------------------------------------------------------
 ## SlicerLibCurl
-find_package(SLICERLIBCURL REQUIRED)
-if(SLICERLIBCURL_FOUND)
-  include(${SLICERLIBCURL_USE_FILE})
-endif(SLICERLIBCURL_FOUND)
+find_package(CURL REQUIRED)
+if(CURL_FOUND)
+  # Add include directories needed to use CURL.
+  include_directories(${CURL_INCLUDE_DIRS})
+endif(CURL_FOUND)
 
 #-----------------------------------------------------------------------------
 ## BatchMake
@@ -317,7 +323,7 @@
 option(Slicer3_USE_OPENIGTLINK "Build Slicer3 with OpenIGTLink support." OFF)
 if(Slicer3_USE_OPENIGTLINK)
   find_package(OpenIGTLink REQUIRED)
-  include(${OpenIGTLink_USE_FILE})
+  include_directories(${OpenIGTLink_INCLUDE_DIRS})
 endif(Slicer3_USE_OPENIGTLINK)
 
 #-----------------------------------------------------------------------------
Index: slicer-3.6.3~svn16592/Libs/vtkITK/vtkITKArchetypeImageSeriesReader.h
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/vtkITK/vtkITKArchetypeImageSeriesReader.h	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/vtkITK/vtkITKArchetypeImageSeriesReader.h	2011-05-09 23:31:20.000000000 -0400
@@ -34,7 +34,7 @@
 #include <string>
 
 #include "itkMetaDataDictionary.h"
-#include "gdcmDictSet.h"        /// access to dictionary
+// #include "gdcmDictSet.h"        /// access to dictionary
 #include "gdcmDict.h"           /// access to dictionary
 #include "gdcmDictEntry.h"      /// access to dictionary
 #include "gdcmGlobal.h"         /// access to dictionary
Index: slicer-3.6.3~svn16592/CMake/Slicer3FindQT.cmake
===================================================================
--- slicer-3.6.3~svn16592.orig/CMake/Slicer3FindQT.cmake	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/CMake/Slicer3FindQT.cmake	2011-05-09 23:31:20.000000000 -0400
@@ -9,37 +9,37 @@
 
 IF(Slicer3_USE_QT)
   SET(minimum_required_qt_version "4.6")
-  IF(VTK_USE_QVTK)
+#  IF(VTK_USE_QVTK)
 
     # Check if QT_QMAKE_EXECUTABLE is set
-    IF(NOT VTK_QT_QMAKE_EXECUTABLE)
-      MESSAGE(FATAL_ERROR "error: There is a problem with your configuration, the variable VTK_QT_QMAKE_EXECUTABLE should be exposed by VTK.")
-    ENDIF()
+#    IF(NOT VTK_QT_QMAKE_EXECUTABLE)
+#      MESSAGE(FATAL_ERROR "error: There is a problem with your configuration, the variable VTK_QT_QMAKE_EXECUTABLE should be exposed by VTK.")
+#    ENDIF()
 
-    SET(QT_QMAKE_EXECUTABLE ${VTK_QT_QMAKE_EXECUTABLE})
+#    SET(QT_QMAKE_EXECUTABLE ${VTK_QT_QMAKE_EXECUTABLE})
     
     FIND_PACKAGE(Qt4)
 
     MESSAGE(STATUS "Configuring Slicer with Qt ${QT_VERSION_MAJOR}.${QT_VERSION_MINOR}.${QT_VERSION_PATCH}")
     
-    IF(QT4_FOUND AND QT_QMAKE_EXECUTABLE)
+#    IF(QT4_FOUND AND QT_QMAKE_EXECUTABLE)
       # Check version, note that ${QT_VERSION_PATCH} could also be used
-      IF(NOT "${QT_VERSION_MAJOR}.${QT_VERSION_MINOR}" STREQUAL "${minimum_required_qt_version}")
-        MESSAGE(FATAL_ERROR "error: Slicer requires Qt >= ${minimum_required_qt_version} -- you cannot use Qt ${QT_VERSION_MAJOR}.${QT_VERSION_MINOR}.${QT_VERSION_PATCH}. You should probably reconfigure VTK.")
-      ENDIF()
+#      IF(NOT "${QT_VERSION_MAJOR}.${QT_VERSION_MINOR}" STREQUAL "${minimum_required_qt_version}")
+#        MESSAGE(FATAL_ERROR "error: Slicer requires Qt >= ${minimum_required_qt_version} -- you cannot use Qt ${QT_VERSION_MAJOR}.${QT_VERSION_MINOR}.${QT_VERSION_PATCH}. You should probably reconfigure VTK.")
+#      ENDIF()
       # Enable modules
       SET(QT_USE_QTNETWORK ON)
       SET(QT_USE_QTTEST ${Slicer3_BUILD_TESTING})
 
       # Includes Qt headers
       INCLUDE(${QT_USE_FILE})
-    ELSE(QT4_FOUND AND QT_QMAKE_EXECUTABLE)
-      MESSAGE(FATAL_ERROR "error: Qt >= ${minimum_required_qt_version} was not found on your system. You probably need to set the QT_QMAKE_EXECUTABLE variable.")
-    ENDIF(QT4_FOUND AND QT_QMAKE_EXECUTABLE)
+#    ELSE(QT4_FOUND AND QT_QMAKE_EXECUTABLE)
+#      MESSAGE(FATAL_ERROR "error: Qt >= ${minimum_required_qt_version} was not found on your system. You probably need to set the QT_QMAKE_EXECUTABLE variable.")
+#    ENDIF(QT4_FOUND AND QT_QMAKE_EXECUTABLE)
     
-  ELSE(VTK_USE_QVTK)
-     MESSAGE("error: VTK was not configured to use QT, you probably need to recompile it with VTK_USE_GUISUPPORT ON, VTK_USE_QVTK ON, DESIRED_QT_VERSION 4 and QT_QMAKE_EXECUTABLE set appropriatly. Note that Qt >= ${minimum_required_qt_version} is *required*")
-  ENDIF(VTK_USE_QVTK)
+#  ELSE(VTK_USE_QVTK)
+#     MESSAGE("error: VTK was not configured to use QT, you probably need to recompile it with VTK_USE_GUISUPPORT ON, VTK_USE_QVTK ON, DESIRED_QT_VERSION 4 and QT_QMAKE_EXECUTABLE set appropriatly. Note that Qt >= ${minimum_required_qt_version} is *required*")
+#  ENDIF(VTK_USE_QVTK)
 ENDIF(Slicer3_USE_QT)
 
 ENDMACRO(Slicer3_SETUP_QT)
Index: slicer-3.6.3~svn16592/Libs/MRMLCLI/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/MRMLCLI/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/MRMLCLI/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -103,7 +103,7 @@
 
 INSTALL(TARGETS ${lib_name}
   RUNTIME DESTINATION bin COMPONENT RuntimeLibraries 
-  LIBRARY DESTINATION lib/${PROJECT_NAME} COMPONENT RuntimeLibraries
+  LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
   ARCHIVE DESTINATION lib/${PROJECT_NAME} COMPONENT Development
   )
  
Index: slicer-3.6.3~svn16592/Utilities/Launcher/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Utilities/Launcher/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Utilities/Launcher/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -39,7 +39,7 @@
   "${CMAKE_CURRENT_SOURCE_DIR}/Slicer3Launchers/${launcher_name}")
 
 SET(launcher_binary
-  "${Slicer3_BINARY_DIR}/Slicer3${CMAKE_EXECUTABLE_SUFFIX}")
+  "${Slicer3_BINARY_DIR}/Slicer3-launcher${CMAKE_EXECUTABLE_SUFFIX}")
 
 # MESSAGE(STATUS "launcher_platform: ${launcher_platform}")
 # MESSAGE(STATUS "launcher_name: ${launcher_name}")
@@ -48,7 +48,7 @@
 
 if(EXISTS "${launcher_source}")
   configure_file("${launcher_source}" "${launcher_binary}" COPYONLY)
-  install(PROGRAMS "${launcher_binary}" DESTINATION ".")
+  install(PROGRAMS "${launcher_binary}" DESTINATION ${Slicer3_INSTALL_BIN_DIR})
 endif(EXISTS "${launcher_source}")
 
 # Should this location change, change Application/GUI/Slicer3.cxx as well
Index: slicer-3.6.3~svn16592/Libs/SlicerExecutionModel/GenerateCLP/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/SlicerExecutionModel/GenerateCLP/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/SlicerExecutionModel/GenerateCLP/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -13,10 +13,10 @@
   include(${ModuleDescriptionParser_USE_FILE})
 endif(ModuleDescriptionParser_FOUND)
 
-find_package(TCLAP REQUIRED)
-if(TCLAP_FOUND)
-  include(${TCLAP_USE_FILE})
-endif(TCLAP_FOUND)
+#find_package(TCLAP REQUIRED)
+#if(TCLAP_FOUND)
+#  include(${TCLAP_USE_FILE})
+#endif(TCLAP_FOUND)
 
 find_package(ITK REQUIRED)
 if(ITK_FOUND)
Index: slicer-3.6.3~svn16592/Libs/SlicerExecutionModel/GenerateCLP/UseGenerateCLP.cmake.in
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/SlicerExecutionModel/GenerateCLP/UseGenerateCLP.cmake.in	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/SlicerExecutionModel/GenerateCLP/UseGenerateCLP.cmake.in	2011-05-09 23:31:20.000000000 -0400
@@ -5,10 +5,10 @@
   message(FATAL_ERROR "Cannot build without ITK. Please set ITK_DIR.")
 endif(ITK_FOUND)
 
-find_package(TCLAP REQUIRED)
-if(TCLAP_FOUND)
-  include(${TCLAP_USE_FILE})
-endif(TCLAP_FOUND)
+#find_package(TCLAP REQUIRED)
+#if(TCLAP_FOUND)
+#  include(${TCLAP_USE_FILE})
+#endif(TCLAP_FOUND)
 
 find_package(ModuleDescriptionParser REQUIRED)
 if(ModuleDescriptionParser_FOUND)
Index: slicer-3.6.3~svn16592/Libs/SlicerExecutionModel/ModuleDescriptionParser/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/SlicerExecutionModel/ModuleDescriptionParser/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/SlicerExecutionModel/ModuleDescriptionParser/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -132,7 +132,7 @@
 
 install(TARGETS ${lib_name} 
   RUNTIME DESTINATION bin COMPONENT RuntimeLibraries
-  LIBRARY DESTINATION lib/${PROJECT_NAME} COMPONENT RuntimeLibraries
+  LIBRARY DESTINATION lib COMPONENT RuntimeLibraries
   ARCHIVE DESTINATION lib/${PROJECT_NAME} COMPONENT Development
   )
 
Index: slicer-3.6.3~svn16592/Modules/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -42,7 +42,7 @@
 #   - ScriptedModule (at run-time for its Tcl package)
 
 SET(modules
-  AtlasCreator
+#  AtlasCreator
   CommandLineModule
   Editor
   Endoscopy
@@ -63,14 +63,14 @@
   Tractography
 # TumorGrowth
   Volumes
-  ChangeTracker # AF: refactoring TumorGrowth...
+#  ChangeTracker # AF: refactoring TumorGrowth...
   VolumeRendering
 #  VolumeRendering1 <- merged into VolumeRendering
 # WFEngineModule  <- can't work with 2.6 and Win32??
 # MRAblation <- not working either
   ModelMirror
   ModelTransform
-  EMSegment
+#  EMSegment
   # DistanceTransformModel <- culled from slicer 3.6
   ClipModel
   Measurements
@@ -82,7 +82,7 @@
 #  ExtractSubvolumeROI <- renamed to CropVolume
   CropVolume
   CollectFiducials
-  PETCTFusion
+#  PETCTFusion
   IGTToolSelector
   )
 
@@ -91,7 +91,7 @@
     ${modules}
     OpenIGTLinkIF
     NeuroNav
-    ProstateNav
+#    ProstateNav
     # IGTNavigation
     # OpenCV
     )
Index: slicer-3.6.3~svn16592/Libs/SlicerExecutionModel/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Libs/SlicerExecutionModel/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Libs/SlicerExecutionModel/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -30,8 +30,8 @@
 ###
 
 
-SETIFEMPTY(TCLAP_DIR ${CMAKE_CURRENT_BINARY_DIR}/tclap CACHE PATH "Location of TCLAPConfig.cmake" FORCE)
-ADD_SUBDIRECTORY(tclap)
+#SETIFEMPTY(TCLAP_DIR ${CMAKE_CURRENT_BINARY_DIR}/tclap CACHE PATH "Location of TCLAPConfig.cmake" FORCE)
+#ADD_SUBDIRECTORY(tclap)
 
 SETIFEMPTY(ModuleDescriptionParser_DIR ${CMAKE_CURRENT_BINARY_DIR}/ModuleDescriptionParser CACHE PATH "Location of ModuleDescriptionParserConfig.cmake" FORCE)
 ADD_SUBDIRECTORY(ModuleDescriptionParser)
Index: slicer-3.6.3~svn16592/Base/GUI/Python/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Base/GUI/Python/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Base/GUI/Python/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -87,6 +87,11 @@
   target_link_libraries( ${lib_name} ${PYTHON_LIBRARIES})
 endif (WIN32 AND PYTHON_DEBUG_LIBRARIES)
 
+# Apply user-defined properties to the library target.
+IF(Slicer3_LIBRARY_PROPERTIES)
+  SET_TARGET_PROPERTIES(${lib_name} PROPERTIES ${Slicer3_LIBRARY_PROPERTIES})
+ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
 install(TARGETS ${lib_name}
   RUNTIME DESTINATION ${Slicer3_INSTALL_BIN_DIR} COMPONENT RuntimeLibraries 
   LIBRARY DESTINATION ${Slicer3_INSTALL_LIB_DIR} COMPONENT RuntimeLibraries
Index: slicer-3.6.3~svn16592/Modules/Measurements/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/Measurements/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/Measurements/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -87,6 +87,13 @@
   ${KWWidgets_LIBRARIES}
   )
 
+# Apply user-defined properties to the library target.
+IF(Slicer3_LIBRARY_PROPERTIES)
+  SET_TARGET_PROPERTIES(${lib_name} PROPERTIES
+    ${Slicer3_LIBRARY_PROPERTIES}
+  )
+ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
 slicer3_install_modules(${lib_name})
 
 # --------------------------------------------------------------------------
Index: slicer-3.6.3~svn16592/Modules/ModelTransform/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/ModelTransform/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/ModelTransform/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -87,6 +87,13 @@
   Volumes
   )
 
+# Apply user-defined properties to the library target.
+IF(Slicer3_LIBRARY_PROPERTIES)
+  SET_TARGET_PROPERTIES(${lib_name} PROPERTIES
+    ${Slicer3_LIBRARY_PROPERTIES}
+  )
+ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
 slicer3_install_modules(${lib_name})
 
 # --------------------------------------------------------------------------
Index: slicer-3.6.3~svn16592/Modules/ModelMirror/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/ModelMirror/CMakeLists.txt	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/ModelMirror/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -88,6 +88,13 @@
   Volumes
   )
 
+# Apply user-defined properties to the library target.
+IF(Slicer3_LIBRARY_PROPERTIES)
+  SET_TARGET_PROPERTIES(${lib_name} PROPERTIES
+    ${Slicer3_LIBRARY_PROPERTIES}
+  )
+ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
 slicer3_install_modules(${lib_name})
 
 # --------------------------------------------------------------------------
Index: slicer-3.6.3~svn16592/Modules/FourDImage/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/FourDImage/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/FourDImage/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -105,6 +105,13 @@
   ${VolumesLib}
   )
 
+# Apply user-defined properties to the library target.
+IF(Slicer3_LIBRARY_PROPERTIES)
+  SET_TARGET_PROPERTIES(${lib_name} PROPERTIES
+    ${Slicer3_LIBRARY_PROPERTIES}
+  )
+ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
 install(TARGETS ${lib_name}
   RUNTIME DESTINATION ${Slicer3_INSTALL_MODULES_BIN_DIR} COMPONENT RuntimeLibraries 
   LIBRARY DESTINATION ${Slicer3_INSTALL_MODULES_LIB_DIR} COMPONENT RuntimeLibraries
Index: slicer-3.6.3~svn16592/Modules/IGTToolSelector/CMakeLists.txt
===================================================================
--- slicer-3.6.3~svn16592.orig/Modules/IGTToolSelector/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
+++ slicer-3.6.3~svn16592/Modules/IGTToolSelector/CMakeLists.txt	2011-05-09 23:31:20.000000000 -0400
@@ -82,6 +82,13 @@
   ${KWWidgets_LIBRARIES}
   )
 
+# Apply user-defined properties to the library target.
+IF(Slicer3_LIBRARY_PROPERTIES)
+  SET_TARGET_PROPERTIES(${lib_name} PROPERTIES
+    ${Slicer3_LIBRARY_PROPERTIES}
+  )
+ENDIF(Slicer3_LIBRARY_PROPERTIES)
+
 install(TARGETS ${lib_name}
   RUNTIME DESTINATION ${Slicer3_INSTALL_MODULES_BIN_DIR} COMPONENT RuntimeLibraries 
   LIBRARY DESTINATION ${Slicer3_INSTALL_MODULES_LIB_DIR} COMPONENT RuntimeLibraries
Index: slicer-3.6.3~svn16592/Base/GUI/Tcl/LoadVolume.tcl
===================================================================
--- slicer-3.6.3~svn16592.orig/Base/GUI/Tcl/LoadVolume.tcl	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Base/GUI/Tcl/LoadVolume.tcl	2011-05-09 23:31:20.000000000 -0400
@@ -1079,6 +1079,7 @@
   }
   set dicomDict ""
   set dicomDictCandidates [list \
+    /usr/lib/Slicer3/gdcm/Dicts/gdcm.dic \
     $::Slicer3_BUILD/include/InsightToolkit/gdcm/Dicts/gdcm.dic \
     $itkBinDir/../../include/InsightToolkit/gdcm/Dicts/gdcm.dic \
     $itkBinDir/../Utilities/gdcm/Dicts/gdcm.dic \
Index: slicer-3.6.3~svn16592/Base/Logic/vtkSlicerTransformLogic.cxx
===================================================================
--- slicer-3.6.3~svn16592.orig/Base/Logic/vtkSlicerTransformLogic.cxx	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Base/Logic/vtkSlicerTransformLogic.cxx	2011-05-09 23:31:20.000000000 -0400
@@ -7,8 +7,8 @@
 
   Program:   3D Slicer
   Module:    $RCSfile: vtkSlicerTransformLogic.cxx,v $
-  Date:      $Date: 2010-07-01 17:27:45 -0400 (Thu, 01 Jul 2010) $
-  Version:   $Revision: 14011 $
+  Date:      $Date: 2010-04-13 15:19:58 +0000 (Tue, 13 Apr 2010) $
+  Version:   $Revision: 12724 $
 
 =========================================================================auto=*/
 
@@ -27,32 +27,7 @@
 
 #include "vtkStringArray.h"
 
-#include "itkAffineTransform.h"
-#include "itkBSplineDeformableTransform.h"
-#include "itkCenteredAffineTransform.h"
-#include "itkCenteredEuler3DTransform.h"
-#include "itkCenteredRigid2DTransform.h"
-#include "itkCenteredSimilarity2DTransform.h"
-#include "itkEuler2DTransform.h"
-#include "itkEuler3DTransform.h"
-#include "itkFixedCenterOfRotationAffineTransform.h"
-#include "itkIdentityTransform.h"
-#include "itkQuaternionRigidTransform.h"
-#include "itkRigid2DTransform.h"
-#include "itkRigid3DPerspectiveTransform.h"
-#include "itkRigid3DTransform.h"
-#include "itkScalableAffineTransform.h"
-#include "itkScaleLogarithmicTransform.h"
-#include "itkScaleSkewVersor3DTransform.h"
-#include "itkScaleTransform.h"
-#include "itkScaleVersor3DTransform.h"
-#include "itkSimilarity2DTransform.h"
-#include "itkTranslationTransform.h"
-#include "itkVersorRigid3DTransform.h"
-#include "itkVersorTransform.h"
-#include "itkTransformFactory.h"
-
-vtkCxxRevisionMacro(vtkSlicerTransformLogic, "$Revision: 14011 $");
+vtkCxxRevisionMacro(vtkSlicerTransformLogic, "$Revision: 12724 $");
 vtkStandardNewMacro(vtkSlicerTransformLogic);
 
 //----------------------------------------------------------------------------
@@ -160,73 +135,3 @@
 {
   return 1;
 }
-
-  // Register transform types with ITK factory
-  // This code is from Applications/CLI/BRAINSTools/BRAINSCommonLib/GenericTransformImage.cxx
-  // We do this in order to register ScaleVersor3DTransform, which is not done
-  // in ITK 3.18. The rest of transforms need to be re-registered when a new
-  // transform is added. TODO: this must be revised in 3.20, hopefully
-  // ScaleVersor3DTransform will be out of Review by then
-  void vtkSlicerTransformLogic::RegisterITKTransforms(){
-    //This is needed in order to read and write ScaleVersor3D TransformTypes.
-    //Hopefully in ITK-3-19 this will become part of the non-review transform types.
-    itk::TransformFactory<itk::ScaleVersor3DTransform<double> >::RegisterTransform ();
-    itk::TransformFactory<itk::ScaleVersor3DTransform<float> >::RegisterTransform ();
-
-    itk::TransformFactory<itk::AffineTransform<double,2> >::RegisterTransform ();
-    itk::TransformFactory<itk::AffineTransform<double,3> >::RegisterTransform ();
-    itk::TransformFactory<itk::BSplineDeformableTransform<double,2,2> >::RegisterTransform ();
-    itk::TransformFactory<itk::BSplineDeformableTransform<double,3,3> >::RegisterTransform ();
-    itk::TransformFactory<itk::CenteredAffineTransform<double,2> >::RegisterTransform ();
-    itk::TransformFactory<itk::CenteredAffineTransform<double,3> >::RegisterTransform ();
-    itk::TransformFactory<itk::CenteredEuler3DTransform<double> >::RegisterTransform ();
-    itk::TransformFactory<itk::CenteredRigid2DTransform < double > >::RegisterTransform();
-    itk::TransformFactory<itk::CenteredSimilarity2DTransform<double> >::RegisterTransform ();
-    itk::TransformFactory<itk::Similarity2DTransform<double> >::RegisterTransform ();
-    itk::TransformFactory<itk::Euler2DTransform<double> >::RegisterTransform ();
-    itk::TransformFactory<itk::Euler3DTransform<double> >::RegisterTransform ();
-    itk::TransformFactory<itk::FixedCenterOfRotationAffineTransform<double> >::RegisterTransform ();
-    itk::TransformFactory<itk::IdentityTransform<double,2> >::RegisterTransform ();
-    itk::TransformFactory<itk::IdentityTransform<double,3> >::RegisterTransform ();
-    itk::TransformFactory<itk::QuaternionRigidTransform<double> >::RegisterTransform ();
-    itk::TransformFactory<itk::Rigid2DTransform<double> >::RegisterTransform ();
-    itk::TransformFactory<itk::Rigid3DPerspectiveTransform<double> >::RegisterTransform ();
-    itk::TransformFactory<itk::Rigid3DTransform<double> >::RegisterTransform ();
-    itk::TransformFactory<itk::ScalableAffineTransform<double> >::RegisterTransform ();
-    itk::TransformFactory<itk::ScaleLogarithmicTransform<double> >::RegisterTransform ();
-    itk::TransformFactory<itk::ScaleSkewVersor3DTransform<double> >::RegisterTransform ();
-    itk::TransformFactory<itk::ScaleTransform<double,2> >::RegisterTransform ();
-    itk::TransformFactory<itk::ScaleTransform<double,2> >::RegisterTransform ();
-    itk::TransformFactory<itk::ScaleTransform<double,3> >::RegisterTransform ();
-    itk::TransformFactory<itk::TranslationTransform<double> >::RegisterTransform ();
-    itk::TransformFactory<itk::VersorRigid3DTransform<double> >::RegisterTransform ();
-    itk::TransformFactory<itk::VersorTransform<double> >::RegisterTransform ();
-    itk::TransformFactory<itk::AffineTransform<float,2> >::RegisterTransform ();
-    itk::TransformFactory<itk::AffineTransform<float,3> >::RegisterTransform ();
-    itk::TransformFactory<itk::BSplineDeformableTransform<float,2,2> >::RegisterTransform ();
-    itk::TransformFactory<itk::BSplineDeformableTransform<float,3,3> >::RegisterTransform ();
-    itk::TransformFactory<itk::CenteredAffineTransform<float,2> >::RegisterTransform ();
-    itk::TransformFactory<itk::CenteredAffineTransform<float,3> >::RegisterTransform ();
-    itk::TransformFactory<itk::CenteredEuler3DTransform<float> >::RegisterTransform ();
-    itk::TransformFactory<itk::CenteredRigid2DTransform < float > >::RegisterTransform();
-    itk::TransformFactory<itk::CenteredSimilarity2DTransform<float> >::RegisterTransform ();
-    itk::TransformFactory<itk::Similarity2DTransform<float> >::RegisterTransform ();
-    itk::TransformFactory<itk::Euler2DTransform<float> >::RegisterTransform ();
-    itk::TransformFactory<itk::Euler3DTransform<float> >::RegisterTransform ();
-    itk::TransformFactory<itk::FixedCenterOfRotationAffineTransform<float> >::RegisterTransform ();
-    itk::TransformFactory<itk::IdentityTransform<float,2> >::RegisterTransform ();
-    itk::TransformFactory<itk::IdentityTransform<float,3> >::RegisterTransform ();
-    itk::TransformFactory<itk::QuaternionRigidTransform<float> >::RegisterTransform ();
-    itk::TransformFactory<itk::Rigid2DTransform<float> >::RegisterTransform ();
-    itk::TransformFactory<itk::Rigid3DPerspectiveTransform<float> >::RegisterTransform ();
-    itk::TransformFactory<itk::Rigid3DTransform<float> >::RegisterTransform ();
-    itk::TransformFactory<itk::ScalableAffineTransform<float> >::RegisterTransform ();
-    itk::TransformFactory<itk::ScaleLogarithmicTransform<float> >::RegisterTransform ();
-    itk::TransformFactory<itk::ScaleSkewVersor3DTransform<float> >::RegisterTransform ();
-    itk::TransformFactory<itk::ScaleTransform<float,2> >::RegisterTransform ();
-    itk::TransformFactory<itk::ScaleTransform<float,2> >::RegisterTransform ();
-    itk::TransformFactory<itk::ScaleTransform<float,3> >::RegisterTransform ();
-    itk::TransformFactory<itk::TranslationTransform<float> >::RegisterTransform ();
-    itk::TransformFactory<itk::VersorRigid3DTransform<float> >::RegisterTransform ();
-    itk::TransformFactory<itk::VersorTransform<float> >::RegisterTransform ();
-  }
Index: slicer-3.6.3~svn16592/Base/Logic/vtkSlicerTransformLogic.h
===================================================================
--- slicer-3.6.3~svn16592.orig/Base/Logic/vtkSlicerTransformLogic.h	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Base/Logic/vtkSlicerTransformLogic.h	2011-05-09 23:31:20.000000000 -0400
@@ -7,8 +7,8 @@
 
   Program:   3D Slicer
   Module:    $RCSfile: vtkSlicerTransformLogic.h,v $
-  Date:      $Date: 2010-07-01 17:27:45 -0400 (Thu, 01 Jul 2010) $
-  Version:   $Revision: 14011 $
+  Date:      $Date: 2010-02-15 21:35:35 +0000 (Mon, 15 Feb 2010) $
+  Version:   $Revision: 12142 $
 
 =========================================================================auto=*/
 
@@ -43,10 +43,6 @@
   /// Read transform from file 
   vtkMRMLTransformNode* AddTransform (const char* filename, vtkMRMLScene *scene);
 
-  // Register transform types with ITK factory
-  // TODO: revisit this after the next (3.20?) release of ITK is out!!!
-  void RegisterITKTransforms();
-
   /// 
   /// Write transform's data to a specified file
   int SaveTransform (const char* filename, vtkMRMLTransformNode *transformNode);
Index: slicer-3.6.3~svn16592/Base/GUI/vtkSlicerTransformsGUI.cxx
===================================================================
--- slicer-3.6.3~svn16592.orig/Base/GUI/vtkSlicerTransformsGUI.cxx	2011-05-09 23:31:19.000000000 -0400
+++ slicer-3.6.3~svn16592/Base/GUI/vtkSlicerTransformsGUI.cxx	2011-05-09 23:31:20.000000000 -0400
@@ -20,11 +20,9 @@
 #include "vtkKWLoadSaveDialog.h"
 #include "vtkKWLoadSaveButton.h"
 
-#include "vtkSmartPointer.h"
-
 //---------------------------------------------------------------------------
 vtkStandardNewMacro (vtkSlicerTransformsGUI );
-vtkCxxRevisionMacro ( vtkSlicerTransformsGUI, "$Revision: 14060 $");
+vtkCxxRevisionMacro ( vtkSlicerTransformsGUI, "$Revision: 13134 $");
 
 
 //---------------------------------------------------------------------------
@@ -38,15 +36,6 @@
     NCIGTLabel = NULL;
     BIRNLabel = NULL;
 
-    // Register transform types with ITK factory
-    // This code is from Applications/CLI/BRAINSTools/BRAINSCommonLib/GenericTransformImage.cxx
-    // We do this in order to register ScaleVersor3DTransform, which is not done
-    // in ITK 3.18. The rest of transforms need to be re-registered when a new
-    // transform is added. TODO: this must be revised in 3.20, hopefully
-    // ScaleVersor3DTransform will be out of Review by then      
-    vtkSmartPointer<vtkSlicerTransformLogic> logic = vtkSmartPointer<vtkSlicerTransformLogic>::New();
-    logic->RegisterITKTransforms();
-
 }
 
 
